src/datasources/jwt/jwt.module.ts

```diff
@@ -30,24 +30,35 @@ function jwtClientFactory() {
           ...rest,
         },
         options.secretOrPrivateKey,
+        { algorithm: options.algorithm ?? 'HS256' },
```

Comment:
Nit: if we want to explicitly state the default here (instead of that of the package), I'd move it to a constant. We should also use it in the other methods, `verify` and `decode`.

Reply:
The sign method needs the default value:
https://github.com/auth0/node-jsonwebtoken/blob/bc28861f1fa981ed9c009e29c044a19760a0b128/sign.js#L98
But the other two don't as they accept an array.

Reply:
Also regarding `const`, since it's the default value I think it's safe to hardcode here.

Reply:
> The sign method needs the default value:
https://github.com/auth0/node-jsonwebtoken/blob/bc28861f1fa981ed9c009e29c044a19760a0b128/sign.js#L98

The default value is set there though?

> Also regarding const, since it's the default value I think it's safe to hardcode here.

If we want to explictly specify it, a constant is much clearer imo.

> But the other two don't as they accept an array.

This was an oversight on my accord. I thought the default value was also assigned for those in the package.

Reply:
I noticed that the default value is also hardcoded in similar cases, like in [jsonwebtoken’s code here](https://github.com/auth0/node-jsonwebtoken/blob/bc28861f1fa981ed9c009e29c044a19760a0b128/sign.js#L98). I think following a similar pattern works well in our context, but let me know if you feel strongly about using a constant

Reply:
Yes, the default value is set there, but it’s much safer to define it in our code. This way, if the default changes in the future, we won’t be affected.

Reply:
However, the other method, verify, handles different scenarios when no default value is provided.
[Source](https://github.com/auth0/node-jsonwebtoken/blob/bc28861f1fa981ed9c009e29c044a19760a0b128/verify.js#L132)

Reply:
Looking back at this, what do you think about assigning the default in the service ([where we do the `secretOrPrivateKey`](https://github.com/safe-global/safe-client-gateway/blob/fc9cb89a0dce9ab682384c64089723186bee3a5a/src/datasources/jwt/jwt.service.ts#L32)).

We should also add a test ensuring the default value is used.

I'll leave whether we use a constant or not to your discretion.

Reply:
@iamacook 
Actually, setting the algorithm should be the very last step, as it’s the default for JWT, not specific to our setup.

Regarding testing, how do you think it could cause an issue? Since it’s hardcoded and stable, any break would indicate a bug in jsonwebtoken, not on our end.

Reply:
@iamacook Please see the new changes
https://github.com/safe-global/safe-client-gateway/pull/2104/commits/24d91e768d39bb33c2e0509891aad32e09312257